packages/components/src/__rc__/Select/hooks/useHasCalculatedListboxPosition.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react'
+
+/** This is a helper util to resolve the focus jumping issue with future Select and FilterSelect.
+ * Due to the floating element's position starting as a negative value on render and then jumping to the correct position, this caused the focus to jump to the top of the page.
+ * This now polls to check if the element's position is stable by comparing the first and last position.
+ */
+export const useHasCalculatedListboxPosition = (ref: React.RefObject<HTMLElement>): boolean => {
+  const [isStable, setIsStable] = useState(false)
+  const [lastYPosition, setLastYPosition] = useState<number | null>(null)
+
+  useEffect(() => {
+    const checkPosition = (): void => {
+      if (ref.current) {
+        const { y } = ref.current.getBoundingClientRect()
+        if (lastYPosition === null) {
+          setLastYPosition(y)
+        } else if (y === lastYPosition && y >= 0) {
+          setIsStable(true)
```

Comment:
Would it make sense, and is possible to clear the interval here as well?

Reply:
Nice spot on this, the interval was re-running constantly while the popover was open. I've updated the interval to call itself only while is position is not stable, so we're going from this:

<img width="428" alt="Screenshot 2025-01-30 at 9 17 39 am" src="https://github.com/user-attachments/assets/ea9dd247-a71a-436a-8b92-fd13630fb462" />

To this:
<img width="262" alt="Screenshot 2025-01-30 at 9 29 34 am" src="https://github.com/user-attachments/assets/6817f6f3-78a8-4276-ac5c-14cd64bfaaaf" />